api/api.go

```diff
@@ -327,3 +328,42 @@ func GetRepoContributors(url string) []Contributor {
 
 	return contributors
 }
+
+func getPackage(npmUrl string) (packageName string) {
+
+	i := strings.Index(npmUrl, "package")
+	return npmUrl[i+len("package")+1 : len(npmUrl)]
+}
+
+func NPMtoGithubUrl(npmUrl string) (githubUrl string) {
```

Comment:
Are we sure this method is allowed? I don't see anything against web scraping for npmjs, but this is what the project description says about scraping Github:

> You must not conduct “web scraping” of GitHub, where you hit the web service with
raw URLs and parse the resulting HTML. Oh, the inefficiency...

If your code is allowed, I'm good with sticking with it. If not, one alternative I found was the [npm-repo](https://docs.npmjs.com/cli/v9/commands/npm-repo) command. I'm not sure how much more efficient it is, but using the command line we could do something like
`npm repo axios --browser false`

to get the response
> axios repo available at the following URL:
>   https://github.com/axios/axios

Reply:
i agree with this. I do think that it goes against what Davis will want.

Reply:
Thanks for this. I wasn't sure if we wanted to have to install npm for this but we know its on the ece machines so there should be no issue?